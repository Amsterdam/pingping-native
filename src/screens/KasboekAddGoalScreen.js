import React from 'react';
import {StyleSheet, View, TextInput} from 'react-native';
import {Content, Container, Icon} from 'native-base';
import SimpleHeader from '../components/header/SimpleHeader';
import ContentLayout from '../components/layout/ContentLayout';
import Title from '../components/typography/Title';
import Body from '../components/typography/Body';
import {appColors} from '../lib/colors';
import Button from '../components/Button';

const styles = StyleSheet.create({
  paper: {
    backgroundColor: '#fff',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'stretch',
    padding: 15,
    margin: 10,
    marginTop: 30,
    borderRadius: 20,
  },
  subTitle: {
    fontSize: 18,
  },
  inputContainer: {
    alignSelf: 'stretch',
    padding: 10,
  },
  input: {
    height: 40,
    alignSelf: 'stretch',
    borderColor: appColors.primary,
    borderWidth: 1,
    borderRadius: 50,
    marginBottom: 10,
    paddingLeft: 10,
    paddingRight: 10,
  },
  button: {margin: 10},
  passwordContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 10,
    paddingRight: 10,
    borderRadius: 50,
    borderColor: '#000',
    paddingBottom: 10,
  },
  icon: {marginRight: 30},
  textInput: {
    flex: 1,
    fontFamily: 'Heavitas',
    height: 50,
    fontSize: 20,
  },
});

const INITIAL_STATE = {
  amount: 0,
  goal: '',
  description: '',
};

const KasboekAddGoalScreen = ({navigation}) => {
  const [state, setState] = React.useState({INITIAL_STATE});

  return (
    <Container>
      <SimpleHeader navigation={navigation} color="white" />
      <ContentLayout>
        <Content>
          <Title>Nieuw goal aanmaken</Title>
          <View>
            <View style={styles.paper}>
              <Title style={styles.subTitle}>Nieuw goal</Title>
              <View style={styles.inputContainer}>
                <View style={styles.passwordContainer}>
                  <Icon
                    name="euro"
                    type="FontAwesome"
                    color="#000"
                    style={styles.icon}
                  />
                  <TextInput
                    style={styles.textInput}
                    onChangeText={(text) => setState({...state, amount: text})}
                    value={state.amount}
                    autoFocus
                    keyboardType={'numeric'}
                  />
                </View>
                <Body>Goal titel:</Body>
                <TextInput
                  style={styles.input}
                  onChangeText={(text) => setState({...state, goal: text})}
                  value={state.goal}
                />
                <Body>Goal beschrijving:</Body>
                <TextInput
                  style={styles.input}
                  onChangeText={(text) =>
                    setState({...state, description: text})
                  }
                  value={state.description}
                />
              </View>
            </View>
            <Button label="Goal opslaan" rounded style={styles.button} />
          </View>
        </Content>
      </ContentLayout>
    </Container>
  );
};

export default KasboekAddGoalScreen;
