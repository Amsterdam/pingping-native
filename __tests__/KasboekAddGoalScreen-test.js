import React from 'react';
import KasboekAddGoalScreen from '../src/screens/KasboekAddGoalScreen';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<KasboekAddGoalScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
