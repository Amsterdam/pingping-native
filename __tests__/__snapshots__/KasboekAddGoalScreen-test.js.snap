// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "#fff",
        }
      }
    >
      <View
        noShadow={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "transparent",
              "borderBottomWidth": 0.5,
              "elevation": 0,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 6,
              "paddingRight": 10,
              "paddingTop": 18,
              "right": 0,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
              "top": 0,
            },
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#fff",
              "justifyContent": "flex-start",
            },
          ]
        }
        transparent={true}
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#007aff",
                    "fontSize": 24,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "paddingTop": 2,
                  },
                  Object {
                    "color": "#000",
                  },
                ],
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            type="AntDesign"
          >
            
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 30,
        "position": "relative",
        "width": 750,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Array [
            Object {
              "padding": undefined,
            },
            undefined,
          ]
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableOnAndroid={false}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollIntoView={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          }
        }
        update={[Function]}
        viewIsInsideTabBar={false}
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "fontFamily": "Heavitas",
                  "fontSize": 30,
                  "textAlign": "left",
                },
              ]
            }
            uppercase={false}
          >
            Nieuw goal aanmaken
          </Text>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#fff",
                  "borderRadius": 20,
                  "elevation": 5,
                  "justifyContent": "center",
                  "margin": 10,
                  "marginTop": 30,
                  "padding": 15,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "fontFamily": "Heavitas",
                      "fontSize": 18,
                      "textAlign": "left",
                    },
                  ]
                }
                uppercase={false}
              >
                Nieuw goal
              </Text>
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "padding": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "#000",
                      "borderRadius": 50,
                      "flexDirection": "row",
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                      "paddingRight": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#000",
                            "fontSize": 30,
                          },
                          Object {
                            "color": "#000",
                            "fontSize": 30,
                          },
                          Object {
                            "marginRight": 30,
                          },
                        ],
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    type="FontAwesome"
                  >
                    
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    autoFocus={true}
                    keyboardType="numeric"
                    onChangeText={[Function]}
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "flex": 1,
                        "fontFamily": "Heavitas",
                        "fontSize": 20,
                        "height": 50,
                      }
                    }
                    underlineColorAndroid="transparent"
                  />
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "fontFamily": "Raleway",
                        "fontSize": 15,
                        "lineHeight": 25,
                        "textAlign": "left",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  Goal titel:
                </Text>
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "alignSelf": "stretch",
                      "borderColor": "#fb9f00",
                      "borderRadius": 50,
                      "borderWidth": 1,
                      "height": 40,
                      "marginBottom": 10,
                      "paddingLeft": 10,
                      "paddingRight": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "fontFamily": "Raleway",
                        "fontSize": 15,
                        "lineHeight": 25,
                        "textAlign": "left",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  Goal beschrijving:
                </Text>
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "alignSelf": "stretch",
                      "borderColor": "#fb9f00",
                      "borderRadius": 50,
                      "borderWidth": 1,
                      "height": 40,
                      "marginBottom": 10,
                      "paddingLeft": 10,
                      "paddingRight": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                />
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              forwardedRef={[Function]}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#fb9f00",
                  "borderBottomWidth": null,
                  "borderColor": false,
                  "borderLeftWidth": null,
                  "borderRadius": 57,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 5,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "margin": 10,
                  "marginBottom": 5,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#fff",
                      "fontFamily": "System",
                      "fontSize": 16.5,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                    },
                    Object {
                      "color": "#fff",
                      "fontFamily": "Heavitas",
                      "fontSize": 14,
                    },
                  ]
                }
                uppercase={false}
              >
                Goal opslaan
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </RCTSafeAreaView>
  </View>
</View>
`;
